app/_how-tos/visualize-llm-metrics-with-grafana.md

```diff
@@ -0,0 +1,282 @@
+              options:
+                mistral_format: openai
+                upstream_url: https://api.mistral.ai/v1/chat/completions
+                input_cost: 0.25
+                output_cost: 0.25
+            route_type: llm/v1/chat
+            logging:
+              log_payloads: true
+              log_statistics: true
+            auth:
+              header_name: Authorization
+              header_value: Bearer ${mistral_api_key}
+            weight: 50
+variables:
+  openai_api_key:
+    value: $OPENAI_API_KEY
+  mistral_api_key:
+    value: $MISTRAL_API_KEY
+{% endentity_examples %}
+
+
+## Enable the Prometheus plugin
+
+Before you configure Prometheus, enable the [Prometheus plugin](/plugins/prometheus/) on {{site.base_gateway}}. In this example, we’ve enabled two types of metrics: status code metrics, and AI metrics which expose detailed performance and usage data for AI-related requests:
+
+{% entity_examples %}
+entities:
+  plugins:
+    - name: prometheus
+      config:
+        status_code_metrics: true
+        ai_metrics: true
+        bandwidth_metrics: true
+        latency_metrics: true
+        upstream_health_metrics: true
+{% endentity_examples %}
+
+## Configure Prometheus
+
+Create a `prometheus.yml` file:
+
+```sh
+touch prometheus.yml
+```
+
+Now, add the following to the `prometheus.yml` file to configure Prometheus to scrape {{site.base_gateway}} metrics:
+
+```yaml
+scrape_configs:
+ - job_name: 'kong'
+   scrape_interval: 5s
+   static_configs:
+     - targets: ['kong-quickstart-gateway:8001']
+```
+{: data-deployment-topology="on-prem" }
+
+```yaml
+scrape_configs:
+ - job_name: 'kong'
+   scrape_interval: 5s
+   static_configs:
+     - targets: ['kong-quickstart-gateway:8100']
+```
+Now, run a Prometheus server, and pass it the configuration file created in the previous step:
+
+```sh
+docker run -d --name kong-quickstart-prometheus \
+  --network=kong-quickstart-net -p 9090:9090 \
+  -v $(PWD)/prometheus.yml:/etc/prometheus/prometheus.yml \
+  prom/prometheus:latest
+```
+
+Prometheus will begin to scrape metrics data from Kong AI Gateway.
+
+## Validate Prometheus configuration
+
+You can validate that the plugin is collecting metrics by generating traffic to the example service.
+
+Run the following to query the collected `kong_ai_llm_requests_total` metric data:
+
+```sh
+curl -s 'localhost:9090/api/v1/query?query=kong_ai_llm_requests_total'
+```
+
+This should return the following response:
+
+```text
+{"status":"success","data":{"resultType":"vector","result":[{"metric":{"__name__":"kong_ai_llm_requests_total","ai_model":"gpt-4.1","ai_provider":"openai","instance":"kong-quickstart-gateway:8001","job":"kong","workspace":"default"},"value":[1754897018.422,"2"]},{"metric":{"__name__":"kong_ai_llm_requests_total","ai_model":"mistral-tiny","ai_provider":"mistral","instance":"kong-quickstart-gateway:8001","job":"kong","workspace":"default"},"value":[1754897018.422,"1"]}]}}%
+```
+{: .no-copy-code }
```

Comment:
```suggestion
```
Since we generate traffic later and currently this example returns nothing (since no traffic is generated yet), I think we can just remove it.